### PR TITLE
Status handling

### PR DESCRIPTION
**Pull request by rokroskar**
Handle the situation where a user tries to re-start a server that is in the process of shutting down. 

todo:

- [x] template app secret in the chart

### CODE REVIEW

**Comment by rokroskar** [comment]
PR is staged, seems to work as expected but please test.

File: src/templates/server_status.html [hunk 1]
```diff
@@ -28,11 +28,26 @@
...
+  <meta http-equiv="refresh" content="2">
+
+  {% elif status == "stopping" %}
+  <h3>The server is in the process of terminating.</h3>
+  <meta http-equiv="refresh" content="2">
+  {% endif %}
 </div>
 
-<meta http-equiv="refresh" content="2">
+<script>
```
Comment (leafty) [inline src/templates/server_status.html]: Invalid html here

Comment (rokroskar) [reply src/templates/server_status.html]: oops! thanks for noticing that!

File: src/notebooks_service.py [hunk 2]
```diff
@@ -92,6 +92,8 @@ def __call__(self, environ, start_response):
 app = Flask(__name__)
 app.wsgi_app = ReverseProxied(app.wsgi_app)
 
+app.secret_key = os.environ.get('NOTEBOOKS_SECRET_KEY', '12345abcde')
```
Comment (ableuler) [inline src/notebooks_service.py]: Maybe provide default through charts, not at the application level.

File: src/templates/server_status.html [hunk 3]
```diff
@@ -28,11 +28,24 @@
 
 <div class="jumbotron text-center">
   <h1>Server for {{namespace}}/{{project}}/{{commit_sha}} status: {{status}}</h1>
-  <p>This page will automatically refresh and redirect you to the server when it is running.</p>
-</div>
+  {% if status == "not found" %}
```
Comment (ableuler) [inline src/templates/server_status.html]: For the moment this is a good improvement, but moving forward, the trick with the previous state in a cookie should be replaced with some server-side state handling (use the existing redis pod?). From a client side, I am just interested in the state of the server, and I should get a response for it without assuming that it was the same client which has triggered a certain action beforehand.

**Pull request merged**